gazelle/internal/swiftcfg/swift_config.go

```diff
@@ -111,14 +111,45 @@ func NewSwiftConfig() *SwiftConfig {
 			"swift_client_proto",
 			"swift_server_proto",
 		},
-		SwiftProtoCompilers: map[string]string{
-			"swift_proto":        "@build_bazel_rules_swift//proto/compilers:swift_proto",
-			"swift_client_proto": "@build_bazel_rules_swift//proto/compilers:swift_client_proto",
-			"swift_server_proto": "@build_bazel_rules_swift//proto/compilers:swift_server_proto",
+		SwiftProtoCompilers: map[string][]string{
+			"swift_proto":        {"@build_bazel_rules_swift//proto/compilers:swift_proto"},
+			"swift_client_proto": {"@build_bazel_rules_swift//proto/compilers:swift_client_proto"},
+			"swift_server_proto": {"@build_bazel_rules_swift//proto/compilers:swift_server_proto"},
 		},
 	}
 }
 
+// We need to deep-copy some fields in the swift config as gazelle walks the directory tree
+// in order to ensure directives only apply to the current and descendent directories,
+// and not sibling directories.
+func DeepCopySwiftConfig(sc *SwiftConfig) *SwiftConfig {
+	csc := &SwiftConfig{}
+	*csc = *sc
+	csc.DefaultModuleNames = copyMap(sc.DefaultModuleNames)
+	csc.SwiftLibraryTags = copySlice(sc.SwiftLibraryTags)
+	csc.GenerateSwiftProtoLibraryGRPCFlavors = copySlice(sc.GenerateSwiftProtoLibraryGRPCFlavors)
+	csc.SwiftProtoCompilers = copyMap(sc.SwiftProtoCompilers)
+	return csc
+}
+
+func copySlice[T any](original []T) []T {
+	copy := make([]T, len(original))
+	for i, v := range original {
+		copy[i] = v
+	}
+
+	return copy
```

Comment:
Can we use [the copy function](https://go.dev/ref/spec#Appending_and_copying_slices) for this?